packages/rest/__tests__/client/AbstractRestClient.test.ts

```diff
@@ -482,4 +484,114 @@ describe("AbstractRestClient tests", () => {
             hostname: "test",
         }), "/resource", [Headers.APPLICATION_JSON], fakeResponseStream);
     });
+
+    it("should return full response when requested", async () => {
+        const emitter = new MockHttpRequestResponse();
+        const requestFnc = jest.fn((options, callback) => {
+            ProcessUtils.nextTick(async () => {
+
+                const newEmit = new MockHttpRequestResponse();
+                callback(newEmit);
+
+                await ProcessUtils.nextTick(() => {
+                    newEmit.emit("data", Buffer.from("Sample data", "utf8"));
+                });
+
+                await ProcessUtils.nextTick(() => {
+                    newEmit.emit("end");
+                });
+            });
+
+            return emitter;
+        });
+
+        (https.request as any) = requestFnc;
+
+        const restOptions: IOptionsFullResponse = {
+            resource: "/resource",
+        };
+
+        // create list of all properties
+        const listOfClientProperties = Object.keys(CLIENT_PROPERTY);
+
+        const data = await RestClient.getExpectFullResponse(new Session({hostname: "test"}), restOptions);
+        listOfClientProperties.forEach((property) => expect(data[`${property}`]).toBeDefined());
```

Comment:
Should we look for the actual data instead of just checking that the given properties are defined?
No change required!

Reply:
Yes, we can have another expect to check the data :) The property check is there to make sure it matches the interface.

Reply:
No need to add it though, 
This serves the purpose of verifying that the right things should be defined? :smile:

Reply:
I just remembered why I took out  `expect(data).toMatchSnapshot();` from those tests. The reason is that the snapshot that is generated on my machine differs a bit from the one generated when running in Jenkins (for some reason..), and the tests are failing because of that :(

Reply:
no worries : )